src/qt/shieldexception.h

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_QT_SHIELDEXCEPTION_H
+#define BITCOIN_QT_SHIELDEXCEPTION_H
+
+#include <util/check.h>
+#include <util/system.h>
+
+#include <QApplication>
+#include <QMetaObject>
+#include <QString>
+
+#include <cassert>
+
+template <typename FunctionType> void shieldException(FunctionType&& f) {
```

Comment:
Move to src/qt/guiutil.h?

Reply:
[Updated](https://github.com/bitcoin/bitcoin/pull/18897#issuecomment-625844423).